aws_cloudtrail_rules/aws_manual_securtygroup_changes.py

```diff
@@ -0,0 +1,46 @@
+from panther_base_helpers import deep_get, pattern_match_list
+
+PROD_ACCOUNT_IDS = {'11111111111111', '112233445566'}
+SG_CHANGE_EVENTS = {
+    'CreateSecurityGroup': {
+        'fields': ['groupName', 'vpcId'],
+        'title': 'New security group [{groupName}] created by {actor}',
+    },
+    'AuthorizeSecurityGroupIngress': {
+        'fields': ['groupId'],
+        'title': 'User {actor} has updated security group [{groupId}]',
+    },
+    'AuthorizeSecurityGroupEgress': {
+        'fields': ['groupId'],
+        'title': 'User {actor} has updated security group [{groupId}]',
+    },
+}
+
+ALLOWED_USER_AGENTS = {
+    '* HashiCorp/?.0 Terraform/*',
+    # 'console.ec2.amazonaws.com',
+    # 'cloudformation.amazonaws.com',
+}
+
+
+def rule(event):
+    return (event.get('eventName') in SG_CHANGE_EVENTS.keys() and
+            event.get('recipientAccountId') in PROD_ACCOUNT_IDS and
+            not pattern_match_list(event.get('userAgent'), ALLOWED_USER_AGENTS))
+
+
+def dedup(event):
+    return ':'.join(
+        deep_get(event, 'requestParameters', name)
+        for name in SG_CHANGE_EVENTS[event.get('eventName')]['fields'])
```

Comment:
I had to re-read this a few times to get a handle on it. I would recommend changing `name` to `field`, here and below, to help follow along. Or a brief comment explaining the logic.